From 7cb10620d3ac48b959198067358d629fe170392a Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Wed, 15 Jan 2020 13:10:32 +0100
Subject: [PATCH] GCC: fix clash of struct xdg_surface with getter xdg_surface

GCC complains about declaration of xdg_surface getter in XDGSurfaceWrapperImpl
clashing with the actual definition of xdg_surface struct. So we rename the
getter to be get_xdg_surface.

Bug: 819294
Change-Id: I37c8183e75fee7ff38707678e4d6411f31d95191
---
 ui/ozone/platform/wayland/host/xdg_popup_wrapper_impl.cc   | 4 ++--
 ui/ozone/platform/wayland/host/xdg_surface_wrapper_impl.cc | 2 +-
 ui/ozone/platform/wayland/host/xdg_surface_wrapper_impl.h  | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/ui/ozone/platform/wayland/host/xdg_popup_wrapper_impl.cc b/ui/ozone/platform/wayland/host/xdg_popup_wrapper_impl.cc
index 0c6683828e74..0cc46ce34944 100644
--- a/ui/ozone/platform/wayland/host/xdg_popup_wrapper_impl.cc
+++ b/ui/ozone/platform/wayland/host/xdg_popup_wrapper_impl.cc
@@ -297,8 +297,8 @@ bool XDGPopupWrapperImpl::InitializeStable(WaylandConnection* connection,
   if (!positioner)
     return false;
 
-  xdg_popup_.reset(xdg_surface_get_popup(xdg_surface_->xdg_surface(),
-                                         parent_xdg_surface->xdg_surface(),
+  xdg_popup_.reset(xdg_surface_get_popup(xdg_surface_->get_xdg_surface(),
+                                         parent_xdg_surface->get_xdg_surface(),
                                          positioner));
   if (!xdg_popup_)
     return false;
diff --git a/ui/ozone/platform/wayland/host/xdg_surface_wrapper_impl.cc b/ui/ozone/platform/wayland/host/xdg_surface_wrapper_impl.cc
index d4078e1bc139..4f2b79e190d6 100644
--- a/ui/ozone/platform/wayland/host/xdg_surface_wrapper_impl.cc
+++ b/ui/ozone/platform/wayland/host/xdg_surface_wrapper_impl.cc
@@ -312,7 +312,7 @@ zxdg_surface_v6* XDGSurfaceWrapperImpl::zxdg_surface() const {
   return zxdg_surface_v6_.get();
 }
 
-xdg_surface* XDGSurfaceWrapperImpl::xdg_surface() const {
+xdg_surface* XDGSurfaceWrapperImpl::get_xdg_surface() const {
   DCHECK(xdg_surface_);
   return xdg_surface_.get();
 }
diff --git a/ui/ozone/platform/wayland/host/xdg_surface_wrapper_impl.h b/ui/ozone/platform/wayland/host/xdg_surface_wrapper_impl.h
index e14956ebe3e7..8c19f5a8ed79 100644
--- a/ui/ozone/platform/wayland/host/xdg_surface_wrapper_impl.h
+++ b/ui/ozone/platform/wayland/host/xdg_surface_wrapper_impl.h
@@ -72,7 +72,7 @@ class XDGSurfaceWrapperImpl : public ShellSurfaceWrapper {
   static void CloseTopLevelV6(void* data,
                               struct zxdg_toplevel_v6* zxdg_toplevel_v6);
 
-  xdg_surface* xdg_surface() const;
+  xdg_surface* get_xdg_surface() const;
   zxdg_surface_v6* zxdg_surface() const;
 
  private:
-- 
2.20.1

